checker/src/org/checkerframework/checker/guieffect/GuiEffectVisitor.java

```diff
@@ -156,7 +156,9 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
         }
 
         Effect callerEffect = atypeFactory.getDeclaredEffect(callerElt);
-        assert (callerEffect.equals(effStack.peek()));
+        // Field initializers inside anonymous inner classes show up with a null current-method ---
+        // the traversal goes straight from the class to the initializer.
+        assert (currentMethods.peek() == null || callerEffect.equals(effStack.peek()));
```

Comment:
Would `currentMethods.empty()` be simpler for the first condition? I hope we don't expect a null value on the stack.